doc/source_list_library.tex

```diff
@@ -1,40 +1,42 @@
 \section*{Appendix: List library}
 
-Those list library functions that are not builtins are pre-declared as follows:
+Those list library functions that are not built-ins are pre-declared as follows:
 
 \begin{lstlisting}
-// is_list recurses down the list and checks that it ends with the empty list []
+// is_list recurses down the list and checks that it ends with the empty list null
 
 function is_list(xs) {
-    return is_empty_list(xs) || (is_pair(xs) && is_list(tail(xs)));
+    return is_null(xs) || (is_pair(xs) && is_list(tail(xs)));
 }
 
 // equal computes the structural equality 
 // over its arguments
 
-function equal(item1, item2){
-    return (is_pair(item1) && is_pair(item2))
-        || (is_empty_list(item1) && is_empty_list(item2))
-        || item1 === item2;
+function equal(x, y){
+    return (is_pair(x) && is_pair(y)) 
+        ? (equal(head(x), head(y)) &&
+           equal(tail(x), tail(y)))
+        : (is_null(x) && is_null(y))
```

Comment:
`is_null` is redundant, since `null === null`. Not sure if it was explicitly included for explanation's sake though.

Also, in Source 2018, `[] !== []` so there was `is_empty_list`. In Source 2019, is it better to use `===` to check for `null` instead?